### DISCUSSION TITLE
Implementing swap method for `List`

### DISCUSSION THREAD

**Discussion by whym1here** [modular/modular#2255] [Ideas]
Currently there are [`_reverse`](https://github.com/modularml/mojo/blob/25347b9dfb4848ac6aec8dd88eb378a449edd699/stdlib/src/collections/list.mojo#L254) and [`insert`](https://github.com/modularml/mojo/pull/2148) functions that implement swap functionality. Why not we make a generic global swap function or a private function for `List` struct? 

If this seems like a good idea then I will create a new issue.

**Comment by soraros** [comment]
There is `from algorithm import swap`. Unfortunately, `algorithm` isn't open yet.

**Reply by whym1here** [reply]
Ok then we can refactor the code when algorithm becomes open.
